src/components/Rating/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { faStar as faStarSolid } from '@fortawesome/free-solid-svg-icons/faStar';
+import { faStar as faStarRegular } from '@fortawesome/free-regular-svg-icons/faStar';
+import {
+	FontAwesomeIcon,
+	FontAwesomeIconProps,
+} from '@fortawesome/react-fontawesome';
+import * as React from 'react';
+import range from 'lodash/range';
+import isNumber from 'lodash/isNumber';
+import asRendition from '../../asRendition';
+import { RenditionSystemProps } from '../../common-types';
+import { useTheme } from '../../hooks/useTheme';
+import { Flex } from '../Flex';
+
+interface StarProps extends Omit<FontAwesomeIconProps, 'icon' | 'color'> {
+	full: boolean;
+}
+
+const Star = ({ full, ...rest }: StarProps) => {
+	const theme = useTheme();
+
+	const iconProps = full
+		? {
+				color: theme.rating.star.color.full,
+				icon: faStarSolid,
+		  }
+		: {
+				color: theme.rating.star.color.empty,
+				icon: faStarRegular,
+		  };
+
+	return <FontAwesomeIcon {...iconProps} {...rest} />;
+};
+
+const isStarFull = (hoverValue: number, newValue: number, value: number) => {
+	if (hoverValue > 0) {
+		return newValue <= hoverValue;
+	}
+
+	return isNumber(value) && newValue <= value;
+};
+
+interface InternalRatingProps {
+	children?: React.ComponentType;
+	value: number;
+	onChange: (newValue: number) => any;
+}
+
+const BaseRating = React.memo(
+	({ value, onChange, ...rest }: InternalRatingProps) => {
+		const [hoverValue, setHoverValue] = React.useState(0);
```

Comment:
This looks pretty hacky, why not just depend on CSS to apply hover effects? What I suggest is have an icon SVG and just apply fill on hover: https://css-tricks.com/the-many-ways-to-change-an-svg-fill-on-hover-and-when-to-use-them

Reply:
If you hover on 4th star, all four stars should be highlighted. How to do that with css?

Reply:
I forgot that all previous stars should be highlighted. You can do something like:
```
<div class="container">
  <a href="#" class="hover">H</a>
  <a href="#" class="hover">H</a>
  <a href="#" class="hover">H</a>
</div>
```

and
```
.container:not(:focus-within):not(:hover) {
  .hover { 
    background: #blue;
  }
}

.hover {
  background: #fad29c;
  &:focus ~ &,
  &:hover ~ & {
    background: #6ef1cd;
  }
}
```
with CSS,but it might become more complicated than a JS solution. I'm fine with it then 👍